cpp/src/copying/contiguous_split.cu

```diff
@@ -296,6 +296,43 @@ __global__ void copy_partitions(uint8_t const** src_bufs,
     buf_info[buf_index].valid_count > 0 ? &buf_info[buf_index].valid_count : nullptr);
 }
 
+/**
+ * @brief Kernel which copies data from multiple source buffers to multiple
+ * destination buffers.
+ *
+ * When doing a contiguous_split on X columns comprising N total internal buffers
+ * with M splits, we end up having to copy N*M source/destination buffer pairs.
+ * These copies are further subdivided into batches to distribute the amount of work
+ * to be done as evenly as possible across the multiprocessors on the device.
+ * This kernel is arranged such that each block copies 1 source/destination pair.
+ *
+ * @param src_bufs Input source buffers
+ * @param dst_bufs Destination buffers
+ * @param buf_info Information on the range of values to be copied for each destination buffer
+ */
+template <int block_size>
+__global__ void copy_partitions(uint8_t const** src_bufs,
```

Comment:
This is maybe splitting hairs a bit, but:

This smells like it could be expressed in terms of the other `copy_partitions` overload.  You could have one `copy_partitions` that takes a lambda which takes buf_index and returns the dst pointer.   For the chunked case, it would always just return `user_buffer`, and in the regular case it would return the buffer indexed by `buf_info[buf_index].dst_buf_index`.   

The only reason I mention is because the call to `copy_buffer` takes a ton of parameters and having it duplicated is opening the door for bugs a bit.

Reply:
I changed this to take a lambda expression. I think it could be improved with `enable_if` but I honestly failed at each attempt. Looking for opinions on how to make it better.

Reply:
This looks good.  Nice to have the one big wrapper function.